Added /orbit

notsceptor opened the pull request:
Instead of changing the player's velocity, the updated functionality now involves applying a native in-game effect.

notsceptor committed:
Added the /orbit command

notsceptor committed:
Added the /orbit command

notsceptor committed:
Added the /orbit command

VideoGameSmash12 commented:
Use this event [https://jd.papermc.io/paper/1.20/org/bukkit/event/entity/EntityPotionEffectEvent.html](https://jd.papermc.io/paper/1.20/org/bukkit/event/entity/EntityPotionEffectEvent.html) to re-apply the effect if it gets removed.

notsceptor committed:
Implemented an event listener that reapplies the effect if the player is orbited and tries to remove it.

ncurran02 commented on src/main/java/dev/plex/extras/command/OrbitCommand.java:
```diff
@@ -0,0 +1,83 @@
+public class OrbitCommand extends PlexCommand
+{
+
+    private static final Map<UUID, Boolean> isOrbitedMap = new HashMap<>();
```
Use list instead of map - according to the code, players in the list are orbited... No need for boolean

notsceptor committed:
Made relevant changes (Blame drunk me)

ncurran02 commented on src/main/java/dev/plex/extras/command/OrbitCommand.java:
```diff
@@ -0,0 +1,82 @@
+        return args.length == 1 && silentCheckPermission(sender, this.getPermission()) ? PlexUtils.getPlayerNameList() : ImmutableList.of();
+    }
+
+    private void startOrbiting(Player player, int strength) {
```
Bracket on new line please

ncurran02 commented on src/main/java/dev/plex/extras/command/OrbitCommand.java:
```diff
@@ -0,0 +1,82 @@
+        isOrbited.add(player.getUniqueId());
+    }
+
+    private void stopOrbiting(Player player) {
```
Bracket on new line please

ncurran02 commented on src/main/java/dev/plex/extras/command/OrbitCommand.java:
```diff
@@ -0,0 +1,82 @@
+        isOrbited.remove(player.getUniqueId());
+    }
+
+    public static boolean isPlayerOrbited(UUID playerId) {
```
Bracket on new line please

ncurran02 commented on src/main/java/dev/plex/extras/command/OrbitCommand.java:
```diff
@@ -0,0 +1,82 @@
+@CommandPermissions(permission = "plex.tfmextras.orbit")
+public class OrbitCommand extends PlexCommand
+{
+
```
Remove new line

ncurran02 commented on src/main/java/dev/plex/extras/listener/OrbitEffectListener.java:
```diff
@@ -0,0 +1,31 @@
+        if (event.getEntity() instanceof Player player)
+        {
+            if ((event.getAction() == EntityPotionEffectEvent.Action.CLEARED || event.getAction() == EntityPotionEffectEvent.Action.REMOVED)
+                    && event.getModifiedType() == PotionEffectType.LEVITATION) {
```
Bracket on new line please

ncurran02 reviewed (changes requested):
Just formatting changes

notsceptor committed:
Formatting changes

ncurran02 reviewed (approved):
LGTM!

Telesphoreo commented on src/main/java/dev/plex/extras/command/OrbitCommand.java:
```diff
@@ -0,0 +1,84 @@
+    @Override
+    public @NotNull List<String> smartTabComplete(@NotNull CommandSender sender, @NotNull String alias, @NotNull String[] args) throws IllegalArgumentException
+    {
+        return args.length == 1 && silentCheckPermission(sender, this.getPermission()) ? PlexUtils.getPlayerNameList() : ImmutableList.of();
```
Incomplete tab completion. Doesn't have "stop"

notsceptor committed:
Altered tab completion to include "stop" argument.
Included the isPlayerOrbited check inside the scheduler to fix premature checking
